### DISCUSSION TITLE
What to do when your project is recognized as a virus / trojan / harmful app

### DISCUSSION THREAD

**Discussion by j-px** [Sketchware-Pro/Sketchware-Pro#901] [General]
Some Android systems now ship a built-in virus scanner and most suddenly started flagging some Sketchware Pro-built projects as harmful.

### Solution
Export your project using the Sign/Export menu. Files that are generated by the Run button when having a project open might false-positively be flagged as harmful by anti-virus scanners. The problem (sorta) lies in using `testkey` by default. Or it's the virus scanners' fault. idrk lol  
![Screenshot_20220605-151931_Sketchware Pro](https://github.com/Sketchware-Pro/Sketchware-Pro/assets/40118124/cb50ddec-c2b6-490d-9d96-f62f3136fd60)

Why exactly anti-virus scanners flag apps signed with it as harmful isn't known (to me). Having been signed with `testkey` doesn't automatically mean it's harmful, and definitely not that all .apk files with that signature are trojans or viruses.

If your project still gets flagged after installing that .apk file generated by the Sign/Export menu, it's up to you to fix it. Probably it's a Local library or some other code in your project.

### Security vendors to ignore
Sometimes that's not enough. There are a few untrustworthy security vendors that often bring up false-positive scan reports, like MaxSecure or Ikarus. As a general rule of thumb, if only a few (1-4) vendors detect "malware" in your project, they're false positives. Contact them and ask them to re-scan your project's .apk file and/or remove the false-positive detection. **Only do this with a signed, exported .apk file though!** See section *Solution* above for more info.

#### Success story
For example, Avast had detected `Android:Evo-gen [Trj]` in my demo project's signed and exported .apk file. I filed a false-positive detection form, waited a day or so, and got a mail that they fixed the issue. It used to have 2 detections on VirusTotal.com, [now it has zero](https://www.virustotal.com/gui/file/e079fc195a6365327845fe2f593cb29db68d7b083cc6afc2bdceb2d3345eaa95).

**Comment by LukeeSs** [comment]
The same happened to me.
All apps created in Sketchware are indicated by the antivirus as containing virus.

Android/HackTool.Spamtool.AE
and
Android/HackTool.Spamtool.AC

Why does it happen?

**Reply by ilyassesalama** [reply]
Can you just try to read what Jbk0 wrote above?

**Reply by LukeeSs** [reply]
I already read and tried to do. But that didn't work.

**Reply by ilyassesalama** [reply]
Well in that case, can you try signing your app using [apk-signer from Play Store](https://play.google.com/store/apps/details?id=com.haibison.apksigner)?

**Reply by LukeeSs** [reply]
Yes, I signed my apk using apksigner, but still is pointed as trojan.
But I managed to publish the app on the Play Store normally. It could be a false positive.

**Reply by j-px** [reply]
@LukeeSs please read the discussion message again, I updated it

**Comment by Official-Mlx** [comment]
> Some Android systems now ship a built-in virus scanner and most suddenly started flagging some Sketchware Pro-built projects as harmful.
> 
> ### Solution
> Export your project using the Sign/Export menu. Files that are generated by the Run button when having a project open might false-positively be flagged as harmful by anti-virus scanners. The problem (sorta) lies in using `testkey` by default. Or it's the virus scanners' fault. idrk lol  
> ![Screenshot_20220605-151931_Sketchware Pro](https://github.com/Sketchware-Pro/Sketchware-Pro/assets/40118124/cb50ddec-c2b6-490d-9d96-f62f3136fd60)
> 
> Why exactly anti-virus scanners flag apps signed with it as harmful isn't known (to me). Having been signed with `testkey` doesn't automatically mean it's harmful, and definitely not that all .apk files with that signature are trojans or viruses.
> 
> If your project still gets flagged after installing that .apk file generated by the Sign/Export menu, it's up to you to fix it. Probably it's a Local library or some other code in your project.
> 
> ### Security vendors to ignore
> Sometimes that's not enough. There are a few untrustworthy security vendors that often bring up false-positive scan reports, like MaxSecure or Ikarus. As a general rule of thumb, if only a few (1-4) vendors detect "malware" in your project, they're false positives. Contact them and ask them to re-scan your project's .apk file and/or remove the false-positive detection. **Only do this with a signed, exported .apk file though!** See section *Solution* above for more info.
> 
> #### Success story
> For example, Avast had detected `Android:Evo-gen [Trj]` in my demo project's signed and exported .apk file. I filed a false-positive detection form, waited a day or so, and got a mail that they fixed the issue. It used to have 2 detections on VirusTotal.com, [now it has zero](https://www.virustotal.com/gui/file/e079fc195a6365327845fe2f593cb29db68d7b083cc6afc2bdceb2d3345eaa95).




**I did this with the first released version, dramatically reduced viruses:**

**Before:** [⚜️ Link to VirusTotal](https://virustotal.com/gui/file/1c2738935ff4d05754006dfca7c962d1ca699f5146e4316085b885223bd73c64)
🧬 Detections: 5 / 61

⛔️ Trustlook
  ╰ Android.PUA.DebugKey
⛔️ Google
  ╰ Detected
⛔️ Avast-Mobile
  ╰ Android:Evo-gen [Trj]
⛔️ BitDefenderFalx
  ╰ Android.Riskware.TestKey.rA
⛔️ Ikarus
  ╰ Trojan.AndroidOS.Agent






**after:** [⚜️ Link to VirusTotal](https://virustotal.com/gui/file/e9832dcef903cf4445f343e4397fea7b5d0eb0d3527c705e5194e1d49ff776eb)

🧬 Detections: 2 / 65

⛔️ Google
  ╰ Detected
⛔️ Ikarus
  ╰ Trojan.AndroidOS.Agent



**as a result, it is suggested to use a personal key to sign apps**

**Reply by neoapps-dev** [reply]
yeah, it's probably because of some Local libraries. Idrk lol
